tests/Unit/SpanTest.php

```diff
@@ -160,7 +160,14 @@ public function testSpanLogMessageAddsExpectedTag()
         $span = $this->createSpan();
         $span->log([Tag::LOG_MESSAGE => self::EXCEPTION_MESSAGE]);
 
-        $this->assertFalse($span->hasError());
+        /* I'm not exactly sure what the intended behavior here is, as this
+         * previously did assertFalse($span->hasError).
+         * However, I recently made it so that when ERROR_MSG is set that the
+         * error field will automatically be set to true. This is relied on for
+         * setting the proper fields on the root span, for instance.
+         *  - Levi Morrison, October 2020
+         */
+        $this->assertTrue($span->hasError());
```

Comment:
`log()` is a method inherited from our OpenTracing-like api at the very beginning of this project. Logs are one of the [types of metadata](https://opentracing.io/docs/overview/tags-logs-baggage/#logs).

I think the [behavior in the code](https://github.com/DataDog/dd-trace-php/blob/master/src/DDTrace/Span.php#L369) is wrong as I don't see why a `message` log should be translated into a `error.msg`.

Moreover, logs can be multiple (for different timestamps), so the way it is handled makes no sense to me.

I would suggest to leave this test as `$this->assertFalse($span->hasError());` and mark the test as expected failure.

In the future, we can either remove this functionality and move it to the OpenTracing adapter or to properly handle it.